### PR TITLE
Valgrind with custom branch

### PR DESCRIPTION
**Pull request by thanodnl**
This builds on #40 to allow providing custom branches and email addresses to report to passed in as arguments. This solves #41.

### CODE REVIEW

File: valgrind/launch-test-instance [hunk 1]
```diff
@@ -41,7 +53,13 @@ valgrind_instance_ip=$(aws ec2 describe-instances \
...
-ssh -o StrictHostKeyChecking=no -i $key_name.pem ubuntu@$valgrind_instance_ip "screen -d -m run-valgrind-tests"
+ssh -o StrictHostKeyChecking=no -i $key_name.pem ubuntu@$valgrind_instance_ip 'REPORT_EMAIL="${REPORT_EMAIL}" POSTGRES_GITREF="${POSTGRES_GITREF}" CITUS_GITREF="${CITUS_GITREF}" srceen -d -m run-valgrind-tests'
+
+echo login:
+echo "  " ssh -o StrictHostKeyChecking=no -i $key_name.pem ubuntu@$valgrind_instance_ip
+echo
 
 #delete the key pair after we are done with tests
-aws ec2 delete-key-pair --key-name $key_name
+echo "please run when done"
```
Comment (byucesoy) [inline valgrind/launch-test-instance]: No need for this. Even if we delete the key pair automatically, the instance created with the specific key pair can be accessible with that key pair.

File: valgrind/run-valgrind-tests [hunk 2]
```diff
@@ -20,31 +31,30 @@ sudo DEBIAN_FRONTEND=noninteractive apt-get install -yq \
...
-export PG_CONFIG=/usr/local/pgsql/bin/pg_config
+export PATH=$HOME/pgsql/bin:$PATH
 
 # download and install PostgreSQL
-git clone -b "REL_10_STABLE" --depth 1 git://git.postgresql.org/git/postgresql.git
+git clone -b "${POSTGRES_GITREF}" --depth 1 git://git.postgresql.org/git/postgresql.git
 cd postgresql/
-./configure --enable-cassert --enable-debug CFLAGS="-ggdb -Og -DUSE_VALGRIND"
+./configure \
+    --prefix=$HOME/pgsql \
```
Comment (byucesoy) [inline valgrind/run-valgrind-tests]: :+1: for prefix builds.

File: valgrind/run-valgrind-tests [hunk 3]
```diff
@@ -20,31 +31,30 @@ sudo DEBIAN_FRONTEND=noninteractive apt-get install -yq \
...
+git clone -b "${CITUS_GITREF}" --depth 1 https://github.com/citusdata/citus.git
 cd citus/
 ./configure
 make clean
-make -j8 -s
-sudo make install
-
-# this is necessary to start tests
-sudo chown ubuntu /usr/local/pgsql/bin/ -R
+make install -j8 -s
```
Comment (byucesoy) [inline valgrind/run-valgrind-tests]: This is forgotten in the original script as well but we can use mjobs here too like above.

Though, I don't remember why we are adding 1 to the nproc.

File: valgrind/launch-test-instance (hunk 1 above)
Comment (thanodnl) [reply valgrind/launch-test-instance]: tru, this was for me during development, will revert back

File: valgrind/run-valgrind-tests (hunk 3 above)
Comment (thanodnl) [reply valgrind/run-valgrind-tests]: Good one, I will replace.

re the adding 1, this is mostly done to utilise cpu better. When you have exactly the number of jobs as cores they are often waiting on the disk as compiling is both IO and cpu intensive. By overbooking the jobs on CPU's they are more busy.

**Pull request merged**